torchdata/datapipes/iter/load/online.py

```diff
@@ -80,18 +70,26 @@ class HTTPReaderIterDataPipe(IterDataPipe[Tuple[str, StreamWrapper]]):
     """
 
     def __init__(
-        self, source_datapipe: IterDataPipe[str], timeout: Optional[float] = None, **kwargs: Optional[Dict[str, Any]]
+        self,
+        source_datapipe: IterDataPipe[str],
+        timeout: Optional[float] = None,
+        skip_on_error: bool = False,
+        **kwargs: Optional[Dict[str, Any]],
     ) -> None:
         self.source_datapipe: IterDataPipe[str] = source_datapipe
         self.timeout = timeout
+        self.skip_on_error = skip_on_error
         self.query_params = kwargs
 
     def __iter__(self) -> Iterator[Tuple[str, StreamWrapper]]:
         for url in self.source_datapipe:
-            if self.query_params:
+            try:
                 yield _get_response_from_http(url, timeout=self.timeout, **self.query_params)
-            else:
-                yield _get_response_from_http(url, timeout=self.timeout)
+            except RequestException as e:
```

Comment:
Question: What makes you decide to catch `RequestException` vs other ones or all of them?

Reply:
Good question. All exceptions explicitly thrown by Requests inherit from RequestException (see [doc](https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions)), but just to be safe I'll change this to `Exception`.